src/atomate2/common/jobs/phonons.py

```diff
@@ -250,115 +249,83 @@ def generate_frequencies_eigenvectors(
 def run_phonon_displacements(
     displacements,
     structure: Structure,
-    supercell_matrix,
-    phonon_maker: BaseVaspMaker | ForceFieldStaticMaker = None,
-    prev_vasp_dir: str | Path = None,
+    supercell_matrix: Matrix3D,
+    phonon_maker: BaseVaspMaker | ForceFieldStaticMaker | BaseAimsMaker,
+    prev_dir: str | Path = None,
+    prev_dir_argname: str = None,
+    socket: bool = False,
 ):
     """
     Run phonon displacements.
 
-    Note, this job will replace itself with N displacement calculations.
+    Note, this job will replace itself with N displacement calculations,
+    or a single socket calculation for all displacements.
 
     Parameters
     ----------
-    displacements
+    displacements: Sequence
+        All displacements to calculate
     structure: Structure object
         Fully optimized structure used for phonon computations.
     supercell_matrix: Matrix3D
         supercell matrix for meta data
-    phonon_maker : .BaseVaspMaker
-        A VaspMaker to use to generate the elastic relaxation jobs.
-    prev_vasp_dir : str or Path or None
-        A previous vasp calculation directory to use for copying outputs.
+    phonon_maker : .BaseVaspMaker or .ForceFieldStaticMaker or .BaseAimsMaker
+        A maker to use to generate dispacement calculations
+    prev_dir: str or Path
+        The previous working directory
+    prev_dir_argname: str
+        argument name for the prev_dir variable
+    socket: bool
```

Comment:
Does it make sense having this int he common phonon maker?

Reply:
In theory the socket can be used for more codes than just FHI-aims, which is why I have it in common now. However, I see your point where at the moment it is an aims specific keyword.

Reply:
This is now done